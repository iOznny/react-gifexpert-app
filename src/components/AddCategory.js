import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    // Inicio del input
    const [inputValue, setInputValue] = useState('');

    // Cambiar valor del input
    const handleInputChange = (e) => {        
        setInputValue(e.target.value);
        console.log('handleInputChange llamado');
    };

    // Prevenir posteo de form y agregar nuevo valor al arreglo
    const handleSubmit = (e) => {                
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            setCategories(cat => [inputValue, ...cat]);
            setInputValue('');
        }
    }

    return (  
        <form onSubmit={ handleSubmit }>
            <p>{ inputValue }</p>
            <input 
                type="text"
                value={ inputValue } 
                onChange={ handleInputChange }
            />
        </form>
    )
}

// Requerir el prop-types
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
