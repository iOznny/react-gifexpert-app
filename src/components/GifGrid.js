import React from 'react';
import PropTypes from 'prop-types';

// Hooks
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs'

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);
    
    return (
        <>    
            <h5>{ category }</h5>    
            { loading && <p className="animate__animated animate__flash">Cargando...</p> }

            <div className="card-grid">
                {
                    images.map(img => (<GifGridItem key={ img.id } { ...img } />))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}