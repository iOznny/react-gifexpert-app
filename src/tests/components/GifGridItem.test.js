import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas <GifGridItem />', () => {
    test('Show componen successful ', () => {
        const wrapper = shallow(<GifGridItem />);
        expect(wrapper).toMatchSnapshot();
    });
});