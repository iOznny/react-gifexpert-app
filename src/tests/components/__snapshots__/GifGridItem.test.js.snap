// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifGridItem /> Show componen successful  1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img />
  <p />
</div>
`;
